src/api/auks/auks_krb5_cred.c

```diff
@@ -124,6 +124,111 @@ auks_krb_is_local_tgt(krb5_principal princ, krb5_data *realm)
 		auks_krb_data_eq(princ->data[1], *realm);
 }
 
+int
+auks_krb_cc_new_unique(char ** fullname_out)
+{
+        int fstatus;
+
+        krb5_context context;
+	krb5_ccache ccache;
+
+	char *ccache_type = NULL;
+
+	/* Initialize KRB5 context */
+	fstatus = krb5_init_context(&context);
+	if (fstatus) {
+		auks_error("Error while initializing KRB5");
+		fstatus = AUKS_ERROR_KRB5_CRED_INIT_CTX;
+		return fstatus;
+	}
+
+	/* Get the default ccache */
+	fstatus = krb5_cc_default(context, &ccache);
+	if (fstatus) {
+	        auks_error("Error while getting default ccache");
+		fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+		goto out;
+	}
+
+	/* Get the default ccache type */
+	ccache_type = krb5_cc_get_type(context, ccache);
+
+	/* Generate a new unique ccache */
+	fstatus = krb5_cc_new_unique(context, ccache_type, NULL, &ccache);
+	if (fstatus) {
+		auks_error("Error while generating new unique ccache of type %s", ccache_type);
+		fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+		goto out;
+	}
+
+
+	fstatus = krb5_cc_get_full_name(context, ccache, fullname_out);
```

Comment:
according to krb5_cc_get_full_name man page, krb5_free_string() must be called on fullname_out when no longer used to free the memory.
As kerberos calls are limited to the scope of this file, it could be good to malloc the string to a traditional char* and call krb5_free_string() directly here on a temporary variable use in krb5_cc_get_full_name.
It should be added to the function description in the header file that the char** must be freed when no longer used to avoid memory leak. It is the responsibility of the caller to do that.